### PR TITLE
Include Custom LESS in import

### LINKED ISSUE #2108: Custom LESS should be included on import

### PR DESCRIPTION
**Pull request by taylortom**
Includes a refactor of content type iteration to make this happen (course & config are now included in contentMap).

Fixes #2108

### CODE REVIEW

File: plugins/output/adapt/importsource.js [hunk 1]
```diff
@@ -462,114 +452,132 @@ function ImportSource(req, done) {
...
-            } else {
-              detachedElementsMap[originalData._id] = type;
-              logger.log('warn', 'Cannot update ' + originalData._id + '._parentId, ' +  originalData._parentId + ' not found in idMap');
-              return cb();
-            }
-          }
-          // define the custom properties and and pluginLocations
-          var genericPropKeys = Object.keys(dbInstance.getModel(type).schema.paths);
-          var customProps = _.pick(data, _.difference(Object.keys(data), genericPropKeys));
+      if(_.isEmpty(customProps)) resolve(data);
```
Comment (tomgreenfield) [inline plugins/output/adapt/importsource.js]: Just checking you intended to remove the `return` here?

Comment (taylortom) [reply plugins/output/adapt/importsource.js]: Probably should still return to be safe.
```suggestion
      if(_.isEmpty(customProps)) return resolve(data);
```

File: plugins/output/adapt/importsource.js [hunk 2]
```diff
@@ -462,114 +452,132 @@ function ImportSource(req, done) {
...
             logger.log('warn', 'Failed to import ' + type + ' ' + (originalData._id || '') + ' ' + error);
             return done();
-          }
-          // Create a courseAssets record if needed
-          createCourseAssets(type, record, function(error){
-            if(error) return done(error, record);
-            done(null, record);
-          });
+          } // Create a courseAssets record if needed
+          createCourseAssets(type, record, () => done(null, record));
```
Comment (lc-thomasberger) [inline plugins/output/adapt/importsource.js]: cb will never return an error. Is this intended? 
```
createCourseAssets(type, record, error => done(error, record));
```

Comment (taylortom) [reply plugins/output/adapt/importsource.js]: Yeah, the original code didn't do this -- will gladly add this if needed. 
@dancgray do you have any opinions?

Comment (dancgray) [reply plugins/output/adapt/importsource.js]: @taylortom I think it should return the error.

Comment (taylortom) [inline plugins/output/adapt/importsource.js]:
```suggestion
          createCourseAssets(type, record, done);
```

Comment (taylortom) [reply plugins/output/adapt/importsource.js]: Just seen the diff with the original, I'm clearly misremembering things!

**Pull request merged**